pkg/cloudproduct/cloudproduct.go

```diff
@@ -33,6 +37,12 @@ type CloudProduct interface {
 	// SyncPodPortsToGameServer runs after a Pod has been assigned to a Node and before we sync
 	// Pod host ports to the GameServer status.
 	SyncPodPortsToGameServer(*agonesv1.GameServer, *corev1.Pod) error
+
+	// ValidateGameServer is called by GameServer.Validate to allow for product specific validation.
+	ValidateGameServer(*agonesv1.GameServer) []metav1.StatusCause
+
+	// PortAllocatorFactory creates a PortAllocator. c.f. gameservers.NewPortAllocator for parameters.
+	PortAllocatorFactory(int32, int32, informers.SharedInformerFactory, externalversions.SharedInformerFactory) portallocator.Interface
```

Comment:
```suggestion
	NewPortAllocator(int32, int32, informers.SharedInformerFactory, externalversions.SharedInformerFactory) portallocator.Interface
```

Reason being, as `PortAllocatorFactory` makes me think I'm getting a Factory, not an actual allocator. Also `New` since it is essentially a constructor.

Reply:
You're right. I was overly factory happy here. :)